sanic/sanic.py

```diff
@@ -71,7 +74,7 @@ def register_middleware(middleware):
             if attach_to == 'request':
                 self.request_middleware.append(middleware)
             if attach_to == 'response':
-                self.response_middleware.append(middleware)
+                self.response_middleware.insert(0, middleware)
```

Comment:
It's just an iterable. You can iterate over it as many times as you like provided you don't pop i.e.

```python
>>> from collections import deque
>>> my_list = deque(['b'])
>>> my_list.append('c')
>>> my_list.appendleft('a')
>>> for i in my_list:
...     print(i)
... 
a
b
c
>>> for i in my_list:
...     print(i)
... 
a
b
c
>>> 
```